pkg/operator/ceph/cluster/watcher.go

```diff
@@ -91,20 +91,18 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 		// If so we don't need to reconcile, this is done to avoid double reconcile on operator restart
 		// Assume the admin key since we are watching for node status to create OSDs
 		clusterInfo := cephclient.AdminClusterInfo(cluster.Namespace)
-		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, nodeName)
+		crushName := cephclient.NormalizeCrushName(nodeName)
+		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, crushName)
 		if err != nil {
 			// If it fails, this might be due to the the operator just starting and catching an add event for that node
-			logger.Debugf("failed to get osds on node %q", nodeName)
-			return false
+			logger.Debugf("failed to get osds on node %q, assume reconcile is necessary", crushName)
+			return true
```

Comment:
I don't think it's quite right to change the node name to `crushName` for this message. We are still messaging the user about the k8s node. We can include in parens the crush name if we wish for more debugging if we wish. However, as I mentioned above, I don't think this is the right code level to normalize the name.